backend/taskController.py

```diff
@@ -0,0 +1,154 @@
+    clas = args.get('class')
+    dueDate = args.get('dueDate')
+    targetDate = args.get('targetDate')
+    remind = args.get('remind')
+    remindDate = args.get('remindDate')
+
+    #finds highest task id
+    new_id = tasks.find().sort([("taskId", -1)]).limit(1)[0]["taskId"]
+    # user = flask_login.current_user
+
+    task = Task(new_id + 1, 1, name, desc, clas, 
+            datetime.strptime(dueDate, '%m-%d-%Y %I:%M %p'), 
+            datetime.strptime(targetDate, '%m-%d-%Y %I:%M %p'), 
+            remind, datetime.strptime(remindDate, '%m-%d-%Y %I:%M %p'))
+    #inserts new task into mongo
+    tasks.insert_one(task.toMongo())
+    #sets return
+    resp = jsonify(success=True)
+    resp.status_code = 201 #created http code
+    return resp
+
+# Delete a task
+@task.route('/Today', methods=['DELETE'])
+@task.route('/List', methods=['DELETE'])
+@task.route('/Calendar', methods=['DELETE'])
+def taskDelete():
+    # if no one logged in
+    if not(current_user.is_authenticated):
+        resp = jsonify(success=False)
+        resp.status_code = 401 #Unauthorized
+        return resp
+    args = request.get_json()
+    taskId = args.get('taskId')
+    task = tasks.find_one({"taskId" : taskId})
+    if len(list(task)) == 0:
+        resp = jsonify(success=False)
+        resp.status_code = 400 #bad request
+        return resp
+        
+    tasks.delete_one({"taskId" : taskId})
+    resp = jsonify(success=True)
+    resp.status_code = 200 #OK
+    return resp
+    
+# Update a task
+@task.route('/Today', methods=['PUT'])
+@task.route('/List', methods=['PUT'])
+@task.route('/Calendar', methods=['PUT'])
+def taskPut():
+    # if no one logged in
+    if not(current_user.is_authenticated):
+        resp = jsonify(success=False)
+        resp.status_code = 401 #Unauthorized
+        return resp
+    args = request.get_json()
+    taskId = args.get('taskId')
+    task = tasks.find_one({"taskId" : taskId})
+    if len(list(task)) == 0:
+        resp = jsonify(success=False)
+        resp.status_code = 400 #bad request
+        return resp
+    
+    name = args.get('name')
+    desc = args.get('desc')
+    clas = args.get('class')
+    dueDate = datetime.strptime(args.get('dueDate'), '%m-%d-%Y %I:%M %p')
+    targetDate = datetime.strptime(args.get('targetDate'), '%m-%d-%Y %I:%M %p')
+    remind = args.get('remind')
+    remindDate = datetime.strptime(args.get('remindDate'), '%m-%d-%Y %I:%M %p')
+    tasks.update_one({"taskId" : taskId},
+                        {"$set": {
+                            "name" : name,
+                            "desc" : desc,
+                            "class" : clas,
+                            "dueDate" : dueDate,
+                            "targetDate" : targetDate,
+                            "remind" : remind,
+                            "remindDate" : remindDate
+                        }})
+    resp = jsonify(success=True)
+    resp.status_code = 200 #OK
+    return resp
+
```

Comment:
Great Job on sorting the tasks required for Today's view.

Reply:
Facts as well as good use of comments.